src/DAL.Infrastructure/Cache/CacheCleaner.cs

```diff
@@ -0,0 +1,32 @@
+﻿using DAL.Events;
+using DAL.Infrastructure.Cache;
+using StackExchange.Redis;
+
+namespace OrderApplication.Cache
+{
+	internal class CacheCleaner : ICacheCleaner
+	{
+		private readonly IDatabase _database;
+
+		public CacheCleaner(IDatabase database)
+		{
+			_database = database;
+		}
+
+		public async Task CleanEventSeatsCache(IList<EventSeat> seats)
+		{
```

Comment:
In general, this class and interface look like a workaround, but not a good solution. Would you create a new method for each new entity? It would be better to implement all the methods for CRUD operations of your cache client and delegate only responsibility to build cache keys we want to work with to a separate class, depending on the entity type

Reply:
Code really becomes more flexible this way.